### PR TITLE
Making weight decay added to the grads in AdamW to be consistent with Adam

### PR DESCRIPTION
**Pull request by scaomath**
Related: #104898.

When doing research in testing different Tikhonov-type regularizers vs weight decay, last week my student found `torch.optim.Adam` gives a weird error for Fourier Neural Operators, which I realized is typical for expanding complex as real and forgot to convert back. Then I found this was fixed in #100973.

However, after the fix, I asked my student to compare the grad produced by different optimizers. `torch.optim.Adam` and `torch.optim.AdamW` gives drastically different results. Then I realized that in `adamw.py`, the `weight_decay` is treated the vanilla way, not as an $L^2$ regularizer like other optimizers do (`sgd.py` or `adam.py`), i.e., the gradient upon retrieving `w.grad` produced the loss function is not $\text{loss}(y, \hat{y}; w) + \alpha\\|w\\|^2$, where $\alpha$ is the `weight_decay` hyper-param, but rather $\text{loss}(y, \hat{y}; w)$, which is inconsistent with other optimizers.

See also: https://discuss.pytorch.org/t/how-pytorch-implement-weight-decay/8436

### CODE REVIEW

**Comment by scaomath** [comment]
Oh I see! Thanks @janeyx99
